service/dap/server.go

```diff
@@ -492,9 +548,9 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 			}
 		}
 
-		s.log.Debugf("building binary at %s", debugbinary)
+		s.log.Debugf("building binary '%s' from '%s' with flags '%v'", debugbinary, program, buildFlags)
 		switch mode {
-		case "debug":
+		case "debug", "replay", "record":
 			err = gobuild.GoBuild(debugbinary, []string{program}, buildFlags)
```

Comment:
@polinasok Do you know if there's an issue with go modules support? I'm unable to build a module with subpackages (I'm seeing this also on master branch so I discard any of my changes at the moment):

```
2021-04-04T16:05:35-03:00 debug layer=dap building binary '/home/razael/DEV/helloworld/__debug_bin' from '/home/razael/DEV/helloworld_modules/main/main.go' with flags ''
../helloworld_modules/main/main.go:4:2: cannot find package "github.com/lggomez/helloworldmodules/sub1" in any of:
	/usr/local/go/src/github.com/lggomez/helloworldmodules/sub1 (from $GOROOT)
	/home/razael/go/src/github.com/lggomez/helloworldmodules/sub1 (from $GOPATH)
../helloworld_modules/main/main.go:5:2: cannot find package "github.com/lggomez/helloworldmodules/sub2" in any of:
	/usr/local/go/src/github.com/lggomez/helloworldmodules/sub2 (from $GOROOT)
	/home/razael/go/src/github.com/lggomez/helloworldmodules/sub2 (from $GOPATH)
2021-04-04T16:05:35-03:00 error layer=dap Failed to launch: Build error: exit status 1
```

The parameters sent seem correct but for some reason it doesn't behave well with the PWD or GOMODULES environment variables

Reply:
I think it's just doing this it can't find go.mod from the current directory. Changing the PWD environment variable doesn't actually do anything and I don't think anything reads $GOMODULES.